tests/test_a11y.py

```diff
@@ -291,3 +291,30 @@ def test_breadcrumb_expansion(page: Page, url_base: str) -> None:
     expect(page.get_by_label("Breadcrumb").get_by_role("list")).to_contain_text(
         "Update Sphinx configuration during the build"
     )
+
+
+@pytest.mark.a11y
+def test_search_as_you_type(page: Page, url_base: str) -> None:
+    """Search-as-you-type feature should support keyboard navigation.
+
+    When the search-as-you-type (inline search results) feature is enabled,
+    pressing Tab after entering a search query should focus the first inline
+    search result.
+    """
+    page.set_viewport_size({"width": 1440, "height": 720})
+    page.goto(urljoin(url_base, "/examples/kitchen-sink/blocks.html"))
+    # Click the search textbox.
+    searchbox = page.locator("css=.navbar-header-items .search-button__default-text")
+    searchbox.click()
+    # Type a search query.
+    query_input = page.locator("css=#pst-search-dialog input[type=search]")
+    expect(query_input).to_be_visible()
+    query_input.type("test")
+    page.wait_for_timeout(301)  # Search execution is debounced for 300 ms.
+    search_results = page.locator("css=#search-results")
+    expect(search_results).to_be_visible()
+    # Navigate into the inline search results.
+    query_input.press("Tab")
+    focused_element_actual_content = page.evaluate("document.activeElement.textContent")
+    focused_element_expected_content = "1. Test of no sidebar"
+    assert focused_element_actual_content == focused_element_expected_content
```

Comment:
this seems slightly fragile in case we (1) change our website content in a way that changes order of search results, or (2) sphinx changes their searchtools.js in a way that changes order of search results.  Maybe not much to be done about that though, and we'll just fix it if/when it breaks... thoughts?

Reply:
I think I can make it more robust like this:

1. Use a CSS selector like `#search-results .search li:first-of-kind` to get the first node in the results
2. Get the `textContent` of the node
3. Use that programmatically retrieved `textContent` as the expected value

Reply:
Fixed now I think, will let you resolve